vanilla-framework/vanilla-framework
# Change name of Code snippet to Code copyable

## Done

- Changed the name of the Code snippet pattern to Code copyable. This includes:
  - Changing the class from `.p-code-snippet` to `.p-code-copyable` (including the action and input). The old classname is now deprecated, to be removed in v2.0 (so it should be 100% backwards compatible at the moment)
  - Updating the url of the example to /examples/patterns/code-copyable/ and adding a redirect from /examples/patterns/code-snippet/ (which required updating to Jekyll 3.5.1 and adding Jekyll redirect plugin)
  - Updating the copy in the docs
- Simplified the sass in Code copyable and made use of new spacing variables + individual icon mixins

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/
- Click on the "Code copyable" link under "Patterns" and check it takes you to http://0.0.0.0:8101/vanilla-framework/examples/patterns/code-copyable/
- Check that the code copyable pattern still copies to clipboard
- Rename some of the markup to `p-code-snippet` and see that it still looks the same (the copy to clipboard function won't work anymore unless you rebuild the whole thing)
- Go to http://0.0.0.0:8101/vanilla-framework/examples/patterns/code-snippet/ and check that it redirects to http://0.0.0.0:8101/vanilla-framework/examples/patterns/code-copyable/
- Run the docs app (`cd docs && ./run`) and go to http://0.0.0.0:8104
- Check that the sidebar nav has "Code copyable", the page also says "Code copyable" and it links to the example at http://0.0.0.0:8101/vanilla-framework/examples/patterns/code-copyable/ (the example won't work until this change has been pushed to master)

## Details

Fixes #1631
